core/api.go

```diff
@@ -250,24 +250,36 @@ type page struct {
 	LastPage bool         `json:"last_page"`
 }
 
-func (a *API) authnHandler(handler http.Handler) http.Handler {
-	auth := authn.NewAPI(a.accessTokens, networkRPCPrefix)
+func AuthHandler(mux *http.ServeMux, handler http.Handler, rDB *raft.Service, accessTokens *accesstoken.CredentialStore, tlsConfig *tls.Config) http.Handler {
+	authenticator := authn.NewAPI(accessTokens, networkRPCPrefix)
+	authorizer := authz.NewAuthorizer(rDB, grantPrefix, policyByRoute)
+
+	var (
+		x509Cert *x509.Certificate
+		err      error
+	)
+	if tlsConfig != nil {
+		// TODO(kr): set Leaf in TLSConfig and use that here.
+		// TODO(tessr): ^ do I keep this TODO?
```

Comment:
Actually, this might be a good opportunity to be a little more specific with the requirements for this function. Instead of taking a whole *tls.Config, maybe it should just take the pkix.Name to trust `internalDN pkix.Name`. What do you think?